pkg/cached/cache_test.go

```diff
@@ -622,6 +622,14 @@ func TestReplaceable(t *testing.T) {
 	}
 }
 
+func TestReplaceableDifferentType(t *testing.T) {
+	replaceable := cached.Replaceable[bool]{}
+	replaceable.Replace(cached.NewSource(func() cached.Result[bool] {
+		return cached.NewResultOK(false, "hash")
+	}))
+	replaceable.Replace(cached.NewResultOK(true, "hash2"))
```

Comment:
maybe test that results are as expected and not just that it doesn't crash?

Reply:
fixed